observability/src/main/java/io/grpc/observability/ObservabilityConfig.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.observability;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import io.grpc.internal.JsonParser;
+import io.grpc.internal.JsonUtil;
+import io.grpc.observabilitylog.v1.GrpcLogRecord;
+import io.grpc.observabilitylog.v1.GrpcLogRecord.EventType;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/** gRPC Observability configuration processor. */
+final class ObservabilityConfig {
+  private static final String CONFIG_ENV_VAR_NAME = "GRPC_CONFIG_OBSERVABILITY";
+
+  private boolean enableCloudLogging = true;
+  private String destinationProjectId = null;
+  private LogFilter[] logFilters;
+  private GrpcLogRecord.EventType[] eventTypes;
+
+  public static class LogFilter {
+    public String pattern;
+    public Integer headerBytes;
+    public Integer messageBytes;
+  }
+
+  ObservabilityConfig() {
+  }
+
+  void read() throws IOException {
+    parse(System.getenv(CONFIG_ENV_VAR_NAME));
+  }
+
+  @SuppressWarnings("unchecked")
+  void parse(String config) throws IOException {
+    checkArgument(config != null, CONFIG_ENV_VAR_NAME + " value is null!");
+    Map<String, ?> map = (Map<String, ?>) JsonParser.parse(config);
+
+    parseLoggingConfig(JsonUtil.getObject(map, "logging_config"));
+  }
+
+  private void parseLoggingConfig(Map<String,?> loggingConfig) {
+    if (loggingConfig != null) {
+      Boolean value = JsonUtil.getBoolean(loggingConfig, "enable_cloud_logging");
+      if (value != null) {
+        enableCloudLogging = value;
+      }
+      destinationProjectId = JsonUtil.getString(loggingConfig, "destination_project_id");
+      List<?> rawList = JsonUtil.getList(loggingConfig, "log_filters");
+      if (rawList != null) {
+        List<Map<String, ?>> logFilters = JsonUtil.checkObjectList(rawList);
```

Comment:
nit: the naming here is a bit confusing as it's the same as the logFilters field at the class level. Maybe call this jsonLogFilters or something to make it clear they are not of the same type?